[dnf5] Rename Solv{Query,Sack} -> Package{Query,Sack}

dmach opened the pull request:
Based on https://github.com/rpm-software-management/libdnf/pull/1198 - it needs to be merged first

j-mracek commented on include/libdnf/rpm/package.hpp:
```diff
@@ -38,280 +38,439 @@ class Goal;
+
+    /// @return Size the RPM package should occupy after installing on disk (RPMTAG_LONGSIZE).
+    ///         The information is always present - it is retrieved from rpmdb if the package is installed or from repodata if the package is available.
+    /// @since 5.0
```
Really 5.0? What about 1.0.0. The current version is 0.65.0. In future we can increase the version.

dmach replied:
Good point. Let's discuss this next time we meet.
I used package version. Maybe it would be worth using API version independent on the package version.

j-mracek commented on include/libdnf/rpm/package.hpp:
```diff
@@ -38,280 +38,439 @@ class Goal;
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.localPkg
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_local_baseurl(DnfPackage * pkg)
+    std::string get_local_path() const;
```
It can somehow return an empty string, but it is not mentioned in description.

j-mracek replied:
What do you think to rename it to `get_local_package_path`?

dmach replied:
I don't think it can return an empty string. If it is possible, it deserves documentation and a test case.
I'm not sure about `get_local_package_path`. There's extra `package` in it, because it's part of the Package class. On the other hand, we have `get_package_size` too. Need to think about it.

j-mracek commented on include/libdnf/rpm/package.hpp:
```diff
@@ -38,280 +38,439 @@ class Goal;
     return id == other.id && sack == other.sack;
 }
 
+
```
Why did you add additional empty lines?

j-mracek commented on dnfdaemon-server/services/repo/repo.cpp:
```diff
@@ -120,7 +120,7 @@ dnfdaemon::KeyValueMap repo_to_map(
                 query.ifilter_repoid(reponames);
                 for (auto pkg : query) {
-                    size += pkg.get_download_size();
+                    size += pkg.get_package_size();
```
The commit also contains renames of methods, but it is not mentioned in commit message. Please may I ask you to fix it?

j-mracek commented on include/libdnf/rpm/package.hpp:
```diff
@@ -120,14 +120,6 @@ class Package {
-    // TODO(dmach): drop and use get_install_size()/get_package_size() instead?
-    // @replaces dnf:dnf/package.py:attribute:Package.size
-    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_size(DnfPackage * pkg)
-    unsigned long long get_size() const;
```
I am not sure whether we can consider it as a good move. As you can see, the logic from public API was moved to Daemon part and probably we will copy it also into microdnf part.

j-mracek replied:
As an alternative we can think about a rename. What do you think?

dmach replied:
This is part of another PR (#1198) as I stated in the description.
I strongly prefer to remove this method due to its ambiguity.

j-mracek replied:
You are correct in principle, but we have a use case that requires it. It means that we need it.

j-mracek commented on bindings/libdnf/rpm.i:
```diff
@@ -49,8 +49,8 @@
 
-%ignore libdnf::rpm::SolvQuery::SolvQuery(SolvQuery && src);
-%include "libdnf/rpm/solv_query.hpp"
+%ignore libdnf::rpm::SolvQuery::PackageQuery(PackageQuery && src);
```
Somehow the name space looks incorrect. Please could you verify it?

dmach replied:
Thanks for catching this, it's most likely incorrect. Let me check and fix it.

j-mracek commented on bindings/libdnf/rpm.i:
```diff
@@ -19,6 +19,7 @@
     #include "libdnf/rpm/config_repo.hpp"
     #include "libdnf/rpm/nevra.hpp"
     #include "libdnf/rpm/package.hpp"
+    #include "libdnf/rpm/package_query.hpp"
```
I suggest that in commit message it is supposed to write a note why the change was created.

j-mracek commented on bindings/libdnf/rpm.i:
```diff
@@ -20,6 +20,7 @@
     #include "libdnf/rpm/nevra.hpp"
     #include "libdnf/rpm/package.hpp"
     #include "libdnf/rpm/package_query.hpp"
+    #include "libdnf/rpm/package_sack.hpp"
```
I see that the commit changes not only name of the class but also additional related parts. May I ask you for an improvement?

Commit:
Rename Solv{Query,Sack} to Package{Query,Sack}

SolvQuery and SolvSack classes were incorrectly named,
because they work with Packages rather than some Solv objects.

dmach commented:
@lukash I've fixed couple of minor issues and also reordered the imports as you requested.
The PR should be good to go (I hope).
